example_test.go

```diff
@@ -180,6 +181,77 @@ Nested:
 	//    - B: 3
 }
 
+func ExampleAttributeEncoder_encode() {
+	// Create a netlink.AttributeEncoder that encodes to the same message
+	// as that decoded by the netlink.AttributeDecoder example.
+	ae, err := netlink.NewAttributeEncoder()
+	if err != nil {
+		log.Fatalf("failed to create attribute encoder: %v", err)
+	}
+
+	// nested is a nestes structure within out.
+	type nested struct {
+		A, B uint32
+	}
+
+	// out is an example structure we will use to pack netlink attributes.
+	type out struct {
+		Number uint16
+		String string
+		Nested nested
+	}
+
+	o := out{
+		Number: 1,
+		String: "hello world",
+		Nested: nested{
+			A: 2,
+			B: 3,
+		},
+	}
+
+	// encodeNested is an example function used to adapt the ae.Do method
+	// to encode an arbitrary structure.
+	//
+	// It is recommended to create functions such as this one at the package
+	// level, but for this example, it is necessary to create a closure to
+	// fully show the usage of netlink.AttributeEncoder.
+	encodeNested := func(n *nested) func() ([]byte, error) {
```

Comment:
We can probably avoid the pointer here for simplicity in the example: it's only necessary for decoding.